lib/spack/spack/test/util/util_url.py

```diff
@@ -271,6 +285,10 @@ def test_url_join_absolute_paths():
     cwd = os.getcwd()
     # ...to work out what resource it points to)
 
+    if sys.platform == "win32":
+        convert_to_posix_path(cwd)
+        cwd = '/' + cwd
```

Comment:
I think it might make sense to encapsulate 

```
cwd = '/' + cwd
```

Into a function like `convert_path_to_url`:


```
def convert_path_to_url(path)
  if is_windows_path: # however you determine that
    return '/' + path
```

Reply:
If this is only used in the tests than I think it's fine where it is, and for general use IMO the method is a little less obvious if for some reason there's an issue with prepending a `/` at some point, as it may not be immediately apparent that that's the operation being preformed by `convert_path_to_url` (I would expect it to add an `'https:'` or the like), particularly if someone is not familiar with this Windows corner case (we've spent quite a bit of time on in in this PR alone) and explicitly having the operation be present and have a comment addressing what's going on. That's just my opinion, if you feel otherwise, as always I'm more than happy to just change it.

Reply:
If this is the only place it is needed, I'll say the line above needs to change from

```
convert_to_posix_path(cwd)
```

to

```
cwd = convert_to_posix_path(cwd)
```

is this logic still needed anywhere after that?

Reply:
I think we have it in a few other places, but mostly in tests, and for a variety of reasons, all but this fully unrelated to URLs.